syntax = "proto3";

package semanticservice; 

service SemanticCheck{
    rpc checkSemantic(Semantic) returns (ServerResponse);
    rpc unitConversion(ConversionRequest) returns (ConversionRequest);
}

message Empty{
}

//json file content or path to it 
message Semantic{
    string semanticDescription = 1;
    string serverName = 2; 
}

//server response with additional comments on what exactly didn't work out
message ServerResponse{
    bool response = 1;
    string description = 2;
    ConversionList conversionList = 3;
}

//Checkup service returns two lists of strings with the keys and values for the unit conversion.
//Can be appended to dictionaries in the conversion service later 
message ConversionList{
    repeated string keys = 1;
    repeated string source = 2;
    repeated string destination = 3;
}

message ConversionRequest{
    //ConversionList destination = 1;
    //Semantic source = 2;
    //repeated string keys = 3;
    //repeated string source = 4;
    //repeated string destination = 5;
    repeated float values = 6;
    ConversionList conversionList = 7;
}
